### PR TITLE
ddl: Better error when check constraint is off

### LINKED ISSUE #52815: The error for disabled check constraints can be improved
## Enhancement

```
mysql> CREATE TABLE t1 (id INT PRIMARY KEY, CHECK (id<10));
Query OK, 0 rows affected, 1 warning (0.01 sec)

mysql> SHOW WARNINGS;
+---------+------+---------------------------------------+
| Level   | Code | Message                               |
+---------+------+---------------------------------------+
| Warning | 1105 | the switch of check constraint is off |
+---------+------+---------------------------------------+
1 row in set (0.00 sec)
```

This isn't user friendly as it doesn't tell you the name of the switch.

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #52815

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: pkg/ddl/ddl_api.go [hunk 1]
```diff
@@ -96,6 +96,8 @@ const (
 	tiflashCheckPendingTablesRetry = 7
 )
 
+var errCheckConstraintIsOff = errors.NewNoStackError("tidb_enable_check_constraint is off")
```
Comment (zimulala) [inline pkg/ddl/ddl_api.go:99]: s/`tidb_enable_check_constraint is off`/`variable.TiDBEnableCheckConstraint+"is off"` ?

**Comment by dveeden** [comment]
@zimulala @fzzf678 PTAL

**Review by fzzf678 — APPROVED** [review]

**Review by zimulala — APPROVED** [review]

**Pull request merged**